lib/hanami/configuration.rb

```diff
@@ -129,9 +129,13 @@ def logger_instance
     # slice, etc.
     setting :slices, default: {}, constructor: :dup.to_proc
 
+    setting :endpoint_namespace, default: "actions"
+
     # TODO: turn this into a richer "source dirs" setting that can support enabling
     # of container component loading as an opt in behvior
-    setting :component_dir_paths, default: %w[actions repositories views]
+    def component_dir_paths
+      [endpoint_namespace, "repositories", "views"]
+    end
```

Comment:
Ugh, it's always awkward when settings depend on each other, but I think this is fine for now. I need to work on this part of the config next anyway. I'll see what we can do.

Reply:
Are we going thru premature DRY?

`endpoint_namespace` is a key prefix for a container.
`"actions"` here is a file system path fragment.

They happen to have the same value, and (given it's Ruby) we use strings for everything. Would we have done the same change if we had `Pathname` in this array=

The values are conceptually linked, but I would leave the raw string here, to have the implementation to refer to the same level of abstraction: an array of strings to represent FS fragments.

Reply:
Yeah, I was actually just about to revisit this and say something similar. I think this should remain a hard-coded list of string paths in which to look for code to load.

_Especially_ given we don't have motivations for the `endpoint_namespace` to be a central config that drives other application behaviours.